src/backend/columnar/columnar_customscan.c

```diff
@@ -806,12 +935,21 @@ FilterPushdownClauses(PlannerInfo *root, RelOptInfo *rel, List *inputClauses)
 		 * there's something we should do with pseudoconstants here.
 		 */
 		if (rinfo->pseudoconstant ||
-			!bms_is_member(rel->relid, rinfo->required_relids) ||
-			!CheckPushdownClause(root, rel, rinfo->clause))
+			!bms_is_member(rel->relid, rinfo->required_relids))
 		{
 			continue;
 		}
 
+		Node *canonicalForm = eval_const_expressions(root, (Node *) rinfo->clause);
+		canonicalForm = (Node *) canonicalize_qual((Expr *) canonicalForm, false);
```

Comment:
Maybe I should add some tests to show the effect of `canonicalize_qual`, other than that, for now I don't see any other changes that needs to be done before merging (cc: @jeff-davis)

Reply:
Yes, please double check that `canonicalize_qual` has not already been called on the input, and that it does something that looks useful.

Reply:
I guess canonicalize_qual isn't doing the thing that I assumed:

```sql
explain select * from pushdown_test t1 where a > 100 and (a < 200 or (a > 300 and a < 500));
┌──────────────────────────────────────────────────────────────────────────────────────────┐
│                                        QUERY PLAN                                        │
├──────────────────────────────────────────────────────────────────────────────────────────┤
│ Custom Scan (ColumnarScan) on pushdown_test t1  (cost=0.00..0.00 rows=1 width=8)         │
│   Filter: ((a > 100) AND ((a < 200) OR ((a > 300) AND (a < 500))))                       │
│   Columnar Projected Columns: a, b                                                       │
│   Columnar Chunk Group Filters: ((a > 100) AND ((a < 200) OR ((a > 300) AND (a < 500)))) │
└──────────────────────────────────────────────────────────────────────────────────────────┘
(4 rows)
``` 

where, I would expect to see something like:
```sql
│   Columnar Chunk Group Filters: ((a > 100) AND ((a < 200) OR (a > 300)) AND ((a < 200) OR (a < 500))) │
```

Reply:
check if it's useful, otherwise, just remove

Reply:
Looks like `subquery_planner` already calls `canonicalize_qual` indirectly, so let me remove that.